### PR TITLE
HistogramData rollout: OpjFile.cpp to vtkDataSetToScaledDataSetTest.h

### LINKED ISSUE #18776: HistogramData rollout: OpjFile.cpp to vtkDataSetToScaledDataSetTest.h
This issue is part of the HistogramData rollout, see #17641.
Algorithms covered in this issue are:

- MantidPlot/src/origin/OPJFile.cpp
- MantidPlot/src/origin/OPJFile.h
- MantidQt/MantidWidgets/src/FitPropertyBrowser.cpp
- MantidQt/MantidWidgets/src/InstrumentView/InstrumentActor.cpp
- MantidQt/MantidWidgets/src/InstrumentView/InstrumentWidgetPickTab.cpp
- MantidQt/MantidWidgets/src/MWView.cpp
- MantidQt/MantidWidgets/src/PreviewPlot.cpp
- MantidQt/MantidWidgets/src/PropertyHandler.cpp
- MantidQt/RefDetectorViewer/src/RefMatrixWSImageView.cpp
- Vates/VatesAPI/test/vtkDataSetToScaledDataSetTest.h

### PR DESCRIPTION
**Pull request by PranavBahuguna**
This PR is part of refactoring effort to use the new interface of the HistogramData module.
See #17641 for details.

The following algorithms were converted to use `HistogramData`:

- MantidQt/MantidWidgets/src/FitPropertyBrowser.cpp
- MantidQt/MantidWidgets/src/InstrumentView/InstrumentActor.cpp
- MantidQt/MantidWidgets/src/InstrumentView/InstrumentWidgetPickTab.cpp
- MantidQt/MantidWidgets/src/PreviewPlot.cpp
- MantidQt/MantidWidgets/src/PropertyHandler.cpp
- MantidQt/RefDetectorViewer/src/RefMatrixWSImageView.cpp

No performance tests were added.

No algorithms show significantly improved performance.

**To test:**

Code review. None of the changed files have associated unit tests.

Fixes #18776.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by KarlPalmen** [comment]
Code review looks OK. I'll test it by loading a workspace and do some unscripted testing on the instrument view.

**Comment by KarlPalmen** [comment]
Instrument viewing test worked as expected.

**Comment by KarlPalmen** [comment]
:shipit:

File: MantidQt/MantidWidgets/src/InstrumentView/InstrumentActor.cpp [hunk 1]
```diff
@@ -610,11 +592,14 @@ void InstrumentActor::sumDetectorsRagged(QList<int> &dets,
...
+    const auto &X = ws->x(0);
+    const auto &Y = ws->y(0);
+    x.assign(X.begin(), X.end());
+    y.assign(Y.begin(), Y.end());
+
     // add the spectra
     for (size_t i = 0; i < nSpec; ++i) {
-      const Mantid::MantidVec &Y = ws->readY(i);
+      const auto &Y = ws->y(i);
       std::transform(y.begin(), y.end(), Y.begin(), y.begin(),
```
Comment (SimonHeybrock) [inline MantidQt/MantidWidgets/src/InstrumentView/InstrumentActor.cpp]: There is `operator+` and `operator+=` for `HistogramY`.

File: MantidQt/MantidWidgets/src/InstrumentView/InstrumentWidgetPickTab.cpp [hunk 2]
```diff
@@ -1441,9 +1418,8 @@ void DetectorPlotController::prepareDataForSumsPlot(int detid,
     }
   }
 
-  if (err) {
+  if (err)
     std::transform(err->begin(), err->end(), err->begin(), Sqrt());
-  }
```
Comment (SimonHeybrock) [inline MantidQt/MantidWidgets/src/InstrumentView/InstrumentWidgetPickTab.cpp]: There is `operator+` and `+=` for `Histogram` (`HistogramMath.h`), that will do the error propagation internally and you can remove a lot of code above.

File: MantidQt/MantidWidgets/src/InstrumentView/InstrumentWidgetPickTab.cpp [hunk 3]
```diff
@@ -1549,11 +1525,11 @@ void DetectorPlotController::prepareDataForIntegralsPlot(
...
-        const Mantid::MantidVec &Y = ws->readY(index);
+        const auto &Y = ws->y(index);
         double sum = std::accumulate(Y.begin() + imin, Y.begin() + imax, 0);
         xymap[xvalue] = sum;
         if (err) {
-          const Mantid::MantidVec &E = ws->readE(index);
+          const auto &E = ws->e(index);
           std::vector<double> tmp(imax - imin);
           // take squares of the errors
           std::transform(E.begin() + imin, E.begin() + imax, E.begin() + imin,
```
Comment (SimonHeybrock) [inline MantidQt/MantidWidgets/src/InstrumentView/InstrumentWidgetPickTab.cpp]: As above, can use `HistogramMath.h`, I think.

**Review by SimonHeybrock — COMMENTED** [review]
Looks correct, but I recommend using math operator that `HistogramData` provides to make life easier:

**Comment by PranavBahuguna** [comment]
Regarding using the operators you mentioned, the nature of how they are being added makes this somewhat difficult to implement. While it is possible to convert from using vectors to `HistogramX/Y`, it adds only the sectopmof the histogram within `imin` and `imax`. I cannot seem to add that section without constructing a new `HistogramX/Y` like so:

```
const auto &Y = ws->y(index);
const auto &YLim += HistogramY(Y.begin() + imin, Y.begin() + imax);
hist.mutableY += YLim;
```

Should I just go ahead with this anyway, or do something else?

**Comment by SimonHeybrock** [comment]
Right, that would probably mean a larger scale refactoring to make this nicer. I guess we should leave it then.

**Pull request merged**